src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            writeToFile(fromFileName, toFileName);
+        } catch (IOException e) {
+            System.out.println("Unknown IOException error occurred while running " + e);
+        }
+    }
+
+    private int[] readFromFile(String fromFileName) throws IOException {
+        File readFile = new File(fromFileName);
+        int supply = 0;
+        int buy = 0;
+        int result = 0;
+        int[] finalResult = new int[3];
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            String readLines = reader.readLine();
+            while (readLines != null) {
+                String[] splitter = readLines.split(",");
+                switch (splitter[0]) {
+                    case "supply":
+                        supply += Integer.parseInt(splitter[1]);
+                        break;
+                    case "buy":
+                        buy += Integer.parseInt(splitter[1]);
+                        break;
+                    default:
+                        break;
+                }
+                readLines = reader.readLine();
+            }
+            result = supply - buy;
+            finalResult[0] = supply;
+            finalResult[1] = buy;
+            finalResult[2] = result;
+            return finalResult;
```

Comment:
The try block should contain the smallest part of the code where a potential exception can occur.